libraries/adaptive-expressions/tests/expressionParser.test.js

```diff
@@ -837,35 +833,49 @@ const scope = {
-                const actualRefs = parsed.references();
-                assertObjectEquals(actualRefs.sort(), data[2].sort());
+        function iterateAndCheckData(source) {
+            for (const data of source) {
+                const input = data[0].toString();
+                console.log(input);
+                var parsed = Expression.parse(input);
+                assert(parsed !== undefined);
+                var {value: actual, error} = parsed.tryEvaluate(scope);
+                assert(error === undefined, `input: ${input}, Has error: ${error}`);
+    
+                let expected = data[1];
+    
+                assertObjectEquals(actual, expected);
+    
+                //Assert ExpectedRefs
+                if (data.length === 3) {
+                    const actualRefs = parsed.references();
+                    assertObjectEquals(actualRefs.sort(), data[2].sort());
+                }
+    
+                //ToString re-parse
+                const newExpr = Expression.parse(parsed.toString());
+                const newActual = newExpr.tryEvaluate(scope).value;
+                assertObjectEquals(actual, newActual);
             }
-
-            //ToString re-parse
-            const newExpr = Expression.parse(parsed.toString());
-            const newActual = newExpr.tryEvaluate(scope).value;
-            assertObjectEquals(actual, newActual);
         }
+
+        // Any tests related to getting the current time should be instantiated in the actual test.
+        // If `new Date()` is called near the end of a time change (end of an hour or end of a day),
+        // this can intermittently result in failures because the test may run at the start of an hour or day.
+        iterateAndCheckData(dataSource);
+        const timeSensitiveData = [
+            ['utcNow(\'MM-DD-YY HH\')', moment(new Date().toISOString()).utc().format('MM-DD-YY HH')],
+            ['getPastTime(1, \'Year\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().subtract(1, 'years').format('MM-DD-YY')],
+            ['getPastTime(1, \'Month\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().subtract(1, 'months').format('MM-DD-YY')],
+            ['getPastTime(1, \'Week\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().subtract(7, 'days').format('MM-DD-YY')],
+            ['getPastTime(1, \'Day\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().subtract(1, 'days').format('MM-DD-YY')],
+            ['getFutureTime(1, \'Year\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().add(1, 'years').format('MM-DD-YY')],
+            ['getFutureTime(1, \'Month\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().add(1, 'months').format('MM-DD-YY')],
+            ['getFutureTime(1, \'Week\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().add(7, 'days').format('MM-DD-YY')],
+            ['getFutureTime(1, \'Day\', \'MM-dd-yy\')', moment(new Date().toISOString()).utc().add(1, 'days').format('MM-DD-YY')],
```

Comment:
Recommendation is to change these tests to instead compare these `'getPastTime()'` and `'getFutureTime()'` to constant times as opposed to creating a new Date each test run.